### ISSUE TITLE
ENH: colorful graphs

### ISSUE THREAD

**Issue by CJ-Wright** [mrocklin/streamz#59]
Is it possible to color each node in graphvis? If so we may want to come up with a color scheme such that every node type has a dedicated color for ease of viewing (and making nice graphs for presentations/publications).

**Comment by mrocklin** [comment]
It is.  I've often found this page to be a helpful reference for the DOT
language: http://www.graphviz.org/doc/info/attrs.html

**Comment by CJ-Wright** [comment]
Do we just stuff color information in the graphviz nodes somewhere and expect graphviz to know what to do?

**Comment by mrocklin** [comment]
Yes.  It looks like we're using the graphviz library.  A quick search
yields this documentation:
https://graphviz.readthedocs.io/en/stable/manual.html#attributes

**Comment by mrocklin** [comment]
You might also find constructing some DOT graphs by hand to be educational
if you haven't done it already.  It's straightforward and a useful tool for
publication.

**Comment by ordirules** [comment]
sounds good. tried it and seems simple (use `style=filled` with `color=clr`). Looks like we can use much more, such as shape.

Did you have any coloring in mind?
I think differentiating stream methods (`map`, `filter` etc) as well as stream subclasses (for example from `Stream` to `DaskStream`) could be nice. Maybe it could be something simple such as color for the methods and shape for the subclasses. Since 'method' might be a little ambiguous, here's an example:
```python
from streamz import Stream
s = Stream()
s2 = s.map(lambda x :x + 1)
from streamz import DaskStream
# now objects are Futures
s3 = DaskStream.scatter(s2)
# still a Future
s4 = s3.map(lambda x : x**2)
s5 = DaskStream.gather(s4)
```
Shape would be extremely useful. For example, running `filter ` on a `DaskStream` would yield a `Stream` (which is unlikely to be intended use). This could be useful for debugging.

What do you think?

**Comment by CJ-Wright** [comment]
I am good with all of those. I'd like to make these (private) attributes of the classes, this way classes which inherent from them/override them would be easy to use. (eg we'd just need to change the shape once for dask streams if they all inherited)
I don't have any particular coloring in mind, we may consider drawing from an MPL color cycle since those are more likely to be colorblind safe.
